### PR TITLE
Coding standard

### PR DESCRIPTION
**Pull request by ajzkk**
## Purpose

Enable Coding standard

## Description

- Enable coding standard (eslint + google styleguid)
- Remove `bluebird` and replace with build in `Promise`
- Update typescript types to replace `bluebird` with build in `Promise`
- Update all packages and fixed bugs
- Update typescript examples with promise
- Remove package-lock.json so that we can test with different version of node.js
- Update workflow to test with different node version `12.x, 14.x, 16.x, 17.x, 18.x, 19.x`
- Update workflow to include `eslint script` to show force coding standard

Screenshot 2023-03-10 at 1 56 18 PM

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -12,15 +12,16 @@ jobs:
 
     strategy:
       matrix:
-        node-version: [10.x, 12.x, 14.x]
+        node-version: [12.x, 14.x, 16.x, 17.x, 18.x, 19.x]
```
Comment (danfowler) [inline .github/workflows/ci.yml:15]: Why different versions from `publish.yml`?

Comment (ajzkk) [reply .github/workflows/ci.yml:15]: I have updated the same. `[12.x, 14.x, 16.x, 17.x, 18.x, 19.x]`

**Review by danfowler — APPROVED** [review]

File: .eslintrc [hunk 2]
```diff
@@ -1,14 +1,27 @@
...
-    "mocha": true
+    "mocha": true,
+    "es6": true
   },
   "rules": {
-    "no-var": "off",
     "max-len": [
       "warn",
-      80
+      150
```
Comment (aashishgurung) [inline .eslintrc:24]: What is the reason to increase the max length from 80 to 150?

Comment (ajzkk) [reply .eslintrc:24]: `return api.get(_makeOptions(options, path('schedules'), data), callback);` 

If `max-len is 80` upper code is saying is very long. but it is not that too long in my opinion. so I set it to `150`

let me know if you have any suggestion for what max-len is good for our codebase.

Comment (aashishgurung) [reply .eslintrc:24]: Let go with 100 for now if 80 is too small number but let's not increase it too much.

**Review by danfowler — APPROVED** [review]

**Review by aashishgurung — APPROVED** [review]

**Pull request merged**